test/extended/dr/machine_recover.go

```diff
@@ -0,0 +1,330 @@
+
+				// TODO: in the future this would be done by a controller automatically
+				framework.Logf("Recreating master %s", replacedMaster.Name)
+				newMaster := replacedMasterMachine.DeepCopy()
+				// The providerID is relied upon by the machine controller to determine a machine
+				// has been provisioned
+				// https://github.com/openshift/cluster-api/blob/c4a461a19efb8a25b58c630bed0829512d244ba7/pkg/controller/machine/controller.go#L306-L308
+				unstructured.SetNestedField(newMaster.Object, "", "spec", "providerID")
+				newMaster.SetName(replacedMaster.Name)
+				newMaster.SetResourceVersion("")
+				newMaster.SetSelfLink("")
+				newMaster.SetUID("")
+				newMaster.SetCreationTimestamp(metav1.NewTime(time.Time{}))
+				// retry until the machine gets created
+				err = wait.PollImmediate(5*time.Second, 10*time.Minute, func() (bool, error) {
+					_, err := ms.Create(newMaster, metav1.CreateOptions{})
+					if errors.IsAlreadyExists(err) {
+						framework.Logf("Waiting for old machine object %s to be deleted so we can create a new one", replacedMaster.Name)
+						return false, nil
+					}
+					if err != nil {
+						return false, err
+					}
+					return true, nil
+				})
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				framework.Logf("Wait for masters to join as nodes and go ready")
+				err = wait.Poll(30*time.Second, 30*time.Minute, func() (done bool, err error) {
+					defer func() {
+						if r := recover(); r != nil {
+							fmt.Println("Recovered from panic", r)
+						}
+					}()
+					nodes, err := oc.AdminKubeClient().CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: "node-role.kubernetes.io/master="})
+					if err != nil {
+						return false, err
+					}
+					ready := countReady(nodes.Items)
+					if ready != len(masters) {
+						framework.Logf("%d master nodes still unready", len(masters)-ready)
+						return false, nil
+					}
+					return true, nil
+				})
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				framework.Logf("Verify recreated master has healthy etcd endpoint")
+				// unhealthyEtcds = getUnhealthyEtcds(certsDir, masters)
+				// o.Expect(len(unhealthyEtcds)).To(o.BeNumerically("==", 0))
+			})
+	},
+	)
+})
+
+func deleteMachineCheckForRole(oc *exutil.CLI, role string) {
+	config, err := framework.LoadConfig()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	dynamicClient := dynamic.NewForConfigOrDie(config)
+	mhc := dynamicClient.Resource(schema.GroupVersionResource{
+		Group:    "machine.openshift.io",
+		Version:  "v1beta1",
+		Resource: "machinehealthchecks",
+	}).Namespace("openshift-machine-api")
+	err = mhc.Delete("test-health-check"+role, &metav1.DeleteOptions{})
+	o.Expect(err).ToNot(o.HaveOccurred())
+}
+
+func createMachineHealthCheckForRole(oc *exutil.CLI, role string) {
+	config, err := framework.LoadConfig()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	dynamicClient := dynamic.NewForConfigOrDie(config)
+	mhc := dynamicClient.Resource(schema.GroupVersionResource{
```

Comment:
not sure if this is right or I should be doing.. `oc.AsAdmin().DynamicClient().Resource(schema.GroupVersionResource{`

Reply:
The kube client from framework is always assumed to be admin